### ISSUE TITLE
The preliminary phase of experiments

### ISSUE THREAD

**Issue by yutian-zhao** [yutian-zhao/deep-learning-for-automated-planning#1]
# Preparation
Setting up STRIPS-HGN.

Setting up virtual environment. Downgrade python to 3.7 because functions like the ones in Timer have been deprecated. 

Install Fast Downward. Download all C++ compilers and packages in Visual Studio, resolve c compiler conflicts with Mingw64. 

- [x] Confirm the correct place to put the Fast Downward folder.

Clone pyperplan. Use William’s modified version. The current version don’t have SearchMetrics. 

- [x] Confirm the correct place to put the pyperplan folder.

# Training
The training scripts in the experiment folder can run with no errors. The observation is that the training for each epoch uses up the maximum training time, although in the last 10 epochs or so the performance doesn’t rank in the top 10. 

- [ ] Further investigation is needed, for example illustrating validation loss to check convergence.

# Evaluation
Evaluation scripts are not provided, I have written my own ones. There are errors when running eval.py, which involve checking if a variable is a class or an instance in json_encoder.py. json_encoder.py has been modified to remove errors.

- [x] Write scripts to analyze data and draw graphs, which are not provided in the repo. 

# Tasks
- [x] Modify the feature mapper such that it takes different goal states into account.
- [x] Modify a-star in pyperplan to collect all the optimal paths to the dequeued nodes.
- [x] Modify scripts to use pyperplan to get training data.
- [x] Test whether the data is correct (optimal).
- [x] Analysis the performance of the model trained on more data.
- [ ] Modify GBFS in pyperplan collect paths which are over-estimated cost. 
- [ ] Write scripts to solve problems with STRIPS-HGN. STRIPS-HGN are re-trained every time a batch of data are collected.
- [ ] Analysis the time to get training data.
